backend/src/core/routes/tests/campaign.routes.test.ts

```diff
@@ -74,6 +75,168 @@ describe('GET /campaigns', () => {
+    for (let i = 1; i <= 3; i++) {
+      expect(resAsc.body.campaigns[i - 1].name).toEqual(`campaign-${i}`)
+    }
+
+    const resDesc = await request(app)
+      .get('/campaigns')
+      .query({ order_by: 'DESC', sort_by: 'created_at' })
+    expect(resDesc.status).toBe(200)
+    expect(resDesc.body.total_count).toEqual(3)
+    for (let i = 1; i <= 3; i++) {
+      expect(resDesc.body.campaigns[i - 1].name).toEqual(
+        `campaign-${3 - i + 1}`
+      )
+    }
+  })
+
+  test('List campaigns order by sent at', async () => {
+    for (let i = 1; i <= 3; i++) {
+      const campaign = await Campaign.create({
+        name: `campaign-${i}`,
+        userId: 1,
+        type: 'SMS',
+        valid: false,
+        protect: false,
+      } as Campaign)
+      await JobQueue.create({
+        campaignId: campaign.id,
+        status: JobStatus.Sending,
+      } as JobQueue)
+    }
+
+    const resSentAsc = await request(app)
+      .get('/campaigns')
+      .query({ order_by: 'ASC', sort_by: 'sent_at' })
+    expect(resSentAsc.status).toBe(200)
+    expect(resSentAsc.body.total_count).toEqual(3)
+    for (let i = 1; i <= 3; i++) {
+      expect(resSentAsc.body.campaigns[i - 1].name).toEqual(`campaign-${i}`)
+    }
+
+    const resSentDesc = await request(app)
+      .get('/campaigns')
+      .query({ order_by: 'DESC', sort_by: 'sent_at' })
+    expect(resSentDesc.status).toBe(200)
+    expect(resSentDesc.body.total_count).toEqual(3)
+    for (let i = 1; i <= 3; i++) {
+      expect(resSentDesc.body.campaigns[i - 1].name).toEqual(
+        `campaign-${3 - i + 1}`
+      )
+    }
+  })
+
+  test('List campaigns filter by mode', async () => {
+    const mode = [ChannelType.SMS, ChannelType.Email, ChannelType.Telegram]
+    for (let i = 1; i <= 3; i++) {
+      await Campaign.create({
+        name: `campaign-${i}`,
+        userId: 1,
+        type: mode[i - 1],
+        valid: false,
+        protect: false,
+      } as Campaign)
+    }
+
+    for (let i = 1; i <= 3; i++) {
+      const res = await request(app)
+        .get('/campaigns')
+        .query({ type: mode[i - 1] })
+      expect(res.status).toBe(200)
+      expect(res.body).toEqual({
+        total_count: 1,
+        campaigns: expect.arrayContaining([
+          expect.objectContaining({ name: `campaign-${i}` }),
+        ]),
+      })
+    }
+  })
+
+  test('List campaigns filter by status', async () => {
+    await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    } as Campaign)
+    const campaign = await Campaign.create({
+      name: 'campaign-2',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    } as Campaign)
+    await JobQueue.create({
+      campaignId: campaign.id,
+      status: JobStatus.Logged,
+    } as JobQueue)
+
+    const resDraft = await request(app)
+      .get('/campaigns')
+      .query({ status: 'Draft' })
```

Comment:
omw out now, but at first glance, `Draft` should be an enum instead of a string?

Reply:
I think the test is relying on the implicit behavior that the default status is Draft. Would be good to note that in a comment for readability!

My own thinking is there's no need for this if it's an obvious property like `created_at` (I'd say even `sent_at` should be thus annotated), but good to have for something more business logic specific!